projects/ngx-datatable/src/lib/components/datatable.component.ts

```diff
@@ -1162,6 +1162,13 @@ export class DatatableComponent implements OnInit, DoCheck, AfterViewInit, After
 
     // Always go to first page when sorting to see the newly sorted data
     this.offset = 0;
+    // Also emit the page object
+    this.page.emit({
+      count: this.count,
+      pageSize: this.pageSize,
+      limit: this.limit,
+      offset: this.offset
+    });
```

Comment:
I would move it below `this.bodyComponent.updateOffsetY(this.offset);`. So internals are computed first. Also consistent with ` onFooterPage(event: any)`.